### ISSUE TITLE
AutoPush need's more control

### ISSUE THREAD

**Issue by SaulDoesCode** [aofei/air#20]
the [``<link>``](https://github.com/aofei/air/blob/cc9ef405101b2e44bd33f751cdc1d7f92a2282b3/response.go#L191) tag has several variations which don't always need pushing for example with ``preload``/``favicon`` pushing can disturb caching and cause needless bandwidth consumption and sometimes in the worst cases duplicate requests, so there should be a bit more specificity and potentially a feature to exclude/omit certain paths from the push list, (immutable/async cached assets assets that are very big)

see [transplacer's](https://github.com/SaulDoesCode/transplacer/blob/05afc0e5dc1c4ae3f5efb0543d952e06efe340a3/transplacer.go#L518) take on this matter.

**Comment by aofei** [comment]
Yeah, you are right. It seems that the `AutoPush` feature needs to be refined. I'll take the time to study it in a few days.

BTW, you made a mistake in the `transplacer`, I'm fixing it, you'll see.

**Comment by aofei** [comment]
Hi @SaulDoesCode,

See my new commit 7c0d0bb460c3421332f4875f27f5a0699e330ed7.

**Comment by SaulDoesCode** [comment]
Awesome, everyday something is getting better. 😄
